js/src/picc.js

```diff
@@ -0,0 +1,1621 @@
+// aight, for polyfilling common JS APIs
+require('aight');
+
+// web components
+require('./components');
+
+var querystring = require('querystring');
+var d3 = require('d3');
+var async = require('async');
+var formdb = require('formdb');
+
+// create the global picc namespace
+var picc = {};
+
+// common error messages
```

Comment:
One suggestion I noticed when reading this file: it's pretty massive. I'd suggest moving things into separate files and requiring them in. Like API and some of the large constants. This could then be added to the namespace:

```
var api = require('./api');
var picc = {};
pic.api = api;
```

I don't think this is strictly necessary at this time, just something I noticed.